tests/e2e/test_execute_async.py

```diff
@@ -53,10 +83,112 @@ def test_cant_get_results_after_cancel(self, long_running_ae: AsyncExecution):
         with pytest.raises(AsyncExecutionException, match="Query was canceled"):
             long_running_ae.get_result()
 
+    def test_get_async_execution_can_check_status(self, long_running_ae: AsyncExecution):
+        query_id, query_secret = str(long_running_ae.query_id), str(
+            long_running_ae.query_secret
+        )
+
+        with self.connection() as conn:
+            ae = conn.get_async_execution(query_id, query_secret)
+            assert ae.is_running
+
+    def test_get_async_execution_can_cancel_across_threads(self, long_running_ae: AsyncExecution):
+        query_id, query_secret = str(long_running_ae.query_id), str(
+            long_running_ae.query_secret
+        )
+
+        def cancel_query_in_separate_thread(query_id, query_secret):
+            with self.connection() as conn:
+                ae = conn.get_async_execution(query_id, query_secret)
+                ae.cancel()
+
+        threading.Thread(
+            target=cancel_query_in_separate_thread, args=(query_id, query_secret)
+        ).start()
+
+        time.sleep(5)
+
+        long_running_ae.poll_for_status()
+        assert long_running_ae.status == AsyncExecutionStatus.CANCELED
+
+    def test_long_ish_query_canary(self, long_ish_ae: AsyncExecution):
+        """This test verifies that on the current endpoint, the LONG_ISH_QUERY requires
+        at least one poll_for_status call before it is finished. If this test fails, it means
+        the SQL warehouse got faster at executing this query and we should increment the value
+        of GT_FIVE_SECONDS_VALUE
+
+        It would be easier to do this if Databricks SQL had a SLEEP() function :/
+        """
+
+        poll_count = 0
+        while long_ish_ae.is_running:
+            time.sleep(1)
+            long_ish_ae.poll_for_status()
+            poll_count += 1
+
+        assert poll_count > 0
+
+    def test_get_async_execution_and_get_results_without_direct_results(
+        self, long_ish_ae: AsyncExecution
+    ):
+        while long_ish_ae.is_running:
+            time.sleep(1)
+            long_ish_ae.poll_for_status()
 
-    def test_staging_operation(self):
```

Comment:
This diff looks wacky here. I'm removing the `test_staging_operation` for the moment.